### ISSUE TITLE
NodeJS course : Shifting from CJS to ESM

### ISSUE THREAD

**Issue by RushilJalal** [TheOdinProject/curriculum#29158]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

ESM has become the norm over the years, gradually phasing out CJS.
I do understand that CJS is still widely used. Thus, I believe we should let the user decide between CJS and ESM syntax for the same code.
We could also approach this by converting all course code from CJS to ESM.
Let me know your views on this:)

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/paths/full-stack-javascript/courses/nodejs

### (Optional) Discord Name

willyWonka

**Comment by MaoShizhong** [comment]
The team is currently in the process of [revamping the Node course](https://github.com/orgs/TheOdinProject/projects/38/views/3) and the CJS/ESM thing is on the agenda but for a future milestone. Note that the things marked "Dump" are just idea dumps and not actual set plans - each milestone gets addressed once the previous has been completed and shipped.

**Comment by wise-king-sullyman** [comment]
@MaoShizhong do you know if the CJS -> ESM migration is actually blocked by anything?

**Comment by MaoShizhong** [comment]
Mainly team discussion and agreement on exactly how we want to approach things, e.g. full ESM, full ESM with a lesson or so on CJS, start CJS then transition to ESM, introduce both then let people use what they wish etc.

That'd primarily be a decision for @01zulfi but I remember past discussion involving "we like ESM, we'll be doing stuff with it but we'll deal with that later". More of the resources/docs we were linking to were using CJS examples. I imagine after milestone 2 is released (APIs/auth), there will be much less reliance on external CJS so a transition to ESM in some capacity will be much simpler.

**Comment by mao-sz** [comment]
Closing as this is something that is planned but needs to be discussed when the relevant revamp milestone is worked on.
